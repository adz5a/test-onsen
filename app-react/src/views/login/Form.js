import React, { Component,  } from "react";
// import {
//     Input,
//     Button,
//     ProgressCircular
// } from "react-onsenui";
import CircularProgress from "material-ui/CircularProgress";
import TextField from "material-ui/TextField";
import RaisedButton from "material-ui/RaisedButton";
import { connect } from "react-redux";
import {
    LOG_USER,
    SIGN_USER
} from "data/user";
import PropTypes from "prop-types";


export class Form extends Component {

    static propTypes = {
        onSignIn: PropTypes.func,
        onLogin: PropTypes.func,
        processing: PropTypes.bool
    };

    componentWillMount () {

        this.onButtonClicked = () => {

            const creds = {
                email: this.form.elements.email.value,
                password: this.form.elements.password.value,
            };
            return creds;

        }


        this.onSignIn = () => {

            if ( typeof this.props.onSignIn === "function" ) {

                this.props.onSignIn(this.onButtonClicked());

            }

        }

        this.onLogin = () => {

            if ( typeof this.props.onLogin === "function" ) {

                this.props.onLogin(this.onButtonClicked());

            }

        }

        this.form = null;
    }

    componentWillUpdate () {}

    render () {

        return (
            <form
                style={{
                    display: "flex",
                    flexDirection: "column",
                    justifyContent: "center",
                    padding: "0 1em 0 1em"
                }}
                ref={ ref => this.form = ref }
            >
                <TextField 
                    name="email"
                    type="email"
                    placeholder="xyz@yolo.gmail.com"
                />
                <TextField 
                    name="password"
                    type="password"
                    placeholder="****"
                />

            <section
                style={{
                    display: "flex",
                    justifyContent: "space-around",
                    marginTop: "2em"
                }}
            >
                <RaisedButton 
                    onClick={this.onLogin}
                >
                    Login
                </RaisedButton>
                <RaisedButton
                    onClick={this.onSignIn}
                >
                    SignIn
                </RaisedButton>
            </section>
            <section
                style={{
                    justifyContent: "center",
                    marginTop: "2em",
                    display: this.props.processing ? "flex" : "none"
                }}
            >
                <CircularProgress size={60}/>
            </section>

        </form>
        );


    }
}

export function mapDispatchToProps ( dispatch ) {

    return {
        onSignIn ( credentials ) {

            return dispatch({
                type: SIGN_USER,
                data: credentials
            });

        },
        onLogin ( credentials ) {

            return dispatch({
                type: LOG_USER,
                data: credentials
            });

        }
    };

}

export function mapStateToProps ( state ) {

    return {

        processing: state.user.processing

    }

}

export const ConnectedForm = connect(
    mapStateToProps,
    mapDispatchToProps
)(Form);
