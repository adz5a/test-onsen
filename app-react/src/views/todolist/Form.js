import React, { 
    Component,
    // PropTypes
} from "react";
import {
    Input,
    Button
} from "react-onsenui";
import { connect } from "react-redux";
import { findDOMNode } from "react-dom";
import PropTypes from "prop-types";
import {
    ADD_TODO
} from "data/todolist";

export class Form extends Component {

    static propTypes = {
        onAdd: PropTypes.func
    };

    componentWillMount () {

        this.inputs = {};
        this.onAdd = () => {

            if ( typeof this.props.onAdd === "function" ) {

                this.props.onAdd({
                    todo: this.inputs.todo.value,
                    date: Date.now()
                });

            }


        };

    }


    render () {

        return (
            <section
                style={{
                    textAlign: "center",
                    padding: "1em 0 1em 0"
                }}
            >
                <Input 
                    type="text"
                    placeholder="a lot of stuff to do"
                    ref={ ref => this.inputs.todo = findDOMNode(ref)}
                />
                <Button 
                    onClick={this.onAdd}
                    type="submit"
                >
                    Add
                </Button>
            </section>
        );

    }

}


export function mapDispatchToProps ( dispatch ) {

    return {
        onAdd ( todo ) {

            return dispatch({
                type: ADD_TODO,
                data: todo
            });

        }
    }

}


export const ConnectedForm = connect(
    null,
    mapDispatchToProps
)(Form);
