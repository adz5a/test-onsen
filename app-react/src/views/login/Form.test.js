import React from 'react';
import ReactDOM from 'react-dom';
import reactTapEventPlugin from "react-tap-event-plugin";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import { Form } from "./Form";
import renderer from "react-test-renderer";

// necessary for the onTap prop
reactTapEventPlugin();


function createNodeMock ( name ) {

    return {

        style: {}

    };

}


it('renders without crashing', () => {
    ReactDOM.render(
        <MuiThemeProvider>
            <Form />
        </MuiThemeProvider>,
        document.createElement("div")
    );
});

it("matches snapshot", () => {


    const tree = renderer.create(
        <MuiThemeProvider>
            <Form />
        </MuiThemeProvider>,
        { createNodeMock }
    ).toJSON();

    expect(tree).toMatchSnapshot();


});
