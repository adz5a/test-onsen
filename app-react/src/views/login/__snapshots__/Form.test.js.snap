// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": "0 1em 0 1em",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "auto",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 48,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <input
      disabled={false}
      id="email-undefined-undefined-9919"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="xyz@yolo.gmail.com"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "WebkitTextFillColor": "rgba(0, 0, 0, 0.87)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="email"
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottom": "solid 1px",
            "borderColor": "#e0e0e0",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottom": "solid 2px",
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "auto",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 48,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <input
      disabled={false}
      id="password-undefined-undefined-14342"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="****"
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "WebkitTextFillColor": "rgba(0, 0, 0, 0.87)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="password"
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottom": "solid 1px",
            "borderColor": "#e0e0e0",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottom": "solid 2px",
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <section
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-around",
        "marginTop": "2em",
      }
    }
  >
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "minWidth": 88,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "#ffffff",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            Login
          </div>
        </div>
      </button>
    </div>
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "minWidth": 88,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "#ffffff",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            SignIn
          </div>
        </div>
      </button>
    </div>
  </section>
  <section
    style={
      Object {
        "display": "none",
        "justifyContent": "center",
        "marginTop": "2em",
      }
    }
  >
    <div
      max={100}
      min={0}
      mode="indeterminate"
      style={
        Object {
          "display": "inline-block",
          "height": 60,
          "muiPrepared": true,
          "position": "relative",
          "width": 60,
        }
      }
      value={0}
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "height": 60,
            "muiPrepared": true,
            "transition": "20s transform linear 0ms",
            "transitionTimingFunction": "linear",
            "width": 60,
          }
        }
      >
        <svg
          style={
            Object {
              "height": 60,
              "muiPrepared": true,
              "position": "relative",
              "width": 60,
            }
          }
          viewBox="0 0 60 60"
        >
          <circle
            cx={30}
            cy={30}
            fill="none"
            r={28.25}
            strokeMiterlimit="20"
            strokeWidth={3.5}
            style={
              Object {
                "muiPrepared": true,
                "stroke": "#00bcd4",
                "strokeLinecap": "round",
                "transition": "1.5s all ease-in-out 0ms",
              }
            }
          />
        </svg>
      </div>
    </div>
  </section>
</form>
`;
